gcc/rust/typecheck/rust-hir-type-check-base.h

```diff
@@ -34,209 +34,285 @@ class TypeCheckBase : public HIR::HIRVisitor
-  virtual void visit (HIR::MacroInvocation &macro_invoc) {}
-  virtual void visit (HIR::MetaItemPath &meta_item) {}
-  virtual void visit (HIR::MetaItemSeq &meta_item) {}
-  virtual void visit (HIR::MetaWord &meta_item) {}
-  virtual void visit (HIR::MetaNameValueStr &meta_item) {}
-  virtual void visit (HIR::MetaListPaths &meta_item) {}
-  virtual void visit (HIR::MetaListNameValueStr &meta_item) {}
-
-  // rust-pattern.h
-  virtual void visit (HIR::LiteralPattern &pattern) {}
-  virtual void visit (HIR::IdentifierPattern &pattern) {}
-  virtual void visit (HIR::WildcardPattern &pattern) {}
-  //  virtual void visit(RangePatternBound& bound) {}
-  virtual void visit (HIR::RangePatternBoundLiteral &bound) {}
-  virtual void visit (HIR::RangePatternBoundPath &bound) {}
-  virtual void visit (HIR::RangePatternBoundQualPath &bound) {}
-  virtual void visit (HIR::RangePattern &pattern) {}
-  virtual void visit (HIR::ReferencePattern &pattern) {}
-  //  virtual void visit(StructPatternField& field) {}
-  virtual void visit (HIR::StructPatternFieldTuplePat &field) {}
-  virtual void visit (HIR::StructPatternFieldIdentPat &field) {}
-  virtual void visit (HIR::StructPatternFieldIdent &field) {}
-  virtual void visit (HIR::StructPattern &pattern) {}
-  //  virtual void visit(TupleStructItems& tuple_items) {}
-  virtual void visit (HIR::TupleStructItemsNoRange &tuple_items) {}
-  virtual void visit (HIR::TupleStructItemsRange &tuple_items) {}
-  virtual void visit (HIR::TupleStructPattern &pattern) {}
-  //  virtual void visit(TuplePatternItems& tuple_items) {}
-  virtual void visit (HIR::TuplePatternItemsMultiple &tuple_items) {}
-  virtual void visit (HIR::TuplePatternItemsRanged &tuple_items) {}
-  virtual void visit (HIR::TuplePattern &pattern) {}
-  virtual void visit (HIR::GroupedPattern &pattern) {}
-  virtual void visit (HIR::SlicePattern &pattern) {}
-
-  // rust-stmt.h
-  virtual void visit (HIR::EmptyStmt &stmt) {}
-  virtual void visit (HIR::LetStmt &stmt) {}
-  virtual void visit (HIR::ExprStmtWithoutBlock &stmt) {}
-  virtual void visit (HIR::ExprStmtWithBlock &stmt) {}
-
-  // rust-type.h
-  virtual void visit (HIR::TraitBound &bound) {}
-  virtual void visit (HIR::ImplTraitType &type) {}
-  virtual void visit (HIR::TraitObjectType &type) {}
-  virtual void visit (HIR::ParenthesisedType &type) {}
-  virtual void visit (HIR::ImplTraitTypeOneBound &type) {}
-  virtual void visit (HIR::TraitObjectTypeOneBound &type) {}
-  virtual void visit (HIR::TupleType &type) {}
-  virtual void visit (HIR::NeverType &type) {}
-  virtual void visit (HIR::RawPointerType &type) {}
-  virtual void visit (HIR::ReferenceType &type) {}
-  virtual void visit (HIR::ArrayType &type) {}
-  virtual void visit (HIR::SliceType &type) {}
-  virtual void visit (HIR::InferredType &type) {}
-  virtual void visit (HIR::BareFunctionType &type) {}
+  virtual void visit (HIR::Token &tok ATTRIBUTE_UNUSED) override {}
```

Comment:
For nitpicky cleanup purposes only, it is probably better to omit the `virtual` keyword, since `override` does not require it to be present to cause the compiler to check it overrides another function and omitting the word makes the line shorter.

Reply:
I am still having issues with the overloaded virtual method is hidden warning so i will do as you say here for now